lib/jsonld-signatures.js

```diff
@@ -19,31 +19,47 @@ const VerificationError = require('./VerificationError');
  * Cryptographically signs the provided document by adding a `proof` section,
  * based on the provided suite and proof purpose.
  *
- * @param {object|string} document - The document to be signed, either a string
- *  URL (resolved to an object via the given `documentLoader`) or a plain
- *  object (JSON-LD document).
+ * @param {object|string} document - The document to be signed, either
```

Comment:
I see here that the `document` can be a document OR a URL.  This PR does not address the case where `document` is a string at all.  Not only will the context not be added properly in that case, but I think `_ensureSuiteContext` breaks if `document` is a string.

I have a suspicion that we have no tests that involve `document` being a string.

`ProofSet().add` on L48 uses the `documentLoader` to dereference the URL as needed.

https://github.com/digitalbazaar/jsonld-signatures/blob/57a9f5b473f501864848fb29737cb96a7ed97fdf/lib/ProofSet.js#L66-L70

So, it seems to me that `_ensureSuiteContext` needs to be moved into `ProofSet().add()` after the existing documentLoader logic or the document loader business needs to move forward in the order of events.

Reply:
And looking deeper into `ProofSet().add()`, maybe `_ensureSuiteContext` belongs back in the suite?

in `suite.createProof`:

https://github.com/digitalbazaar/jsonld-signatures/blob/57a9f5b473f501864848fb29737cb96a7ed97fdf/lib/ProofSet.js#L80-L83

Reply:
I don't think we support the document being a string. I don't think we ever have. We should remove that.